### PR TITLE
Fix role assignment for user auth

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Manually tested:
- Create with new SP
- Create with cached SP
- Create with user auth

### CODE REVIEW

File: eng/common/TestResources/New-TestResources.ps1 [hunk 1]
```diff
@@ -702,19 +702,19 @@ try {
...
-        if ($principalOwnerAssignment.RoleDefinitionName -eq 'Owner') {
-            Write-Verbose "Successfully assigned ownership of '$ResourceGroupName' to the Test Application '$TestApplicationId'"
+        Log "Attempting to assign the 'Owner' role for '$ResourceGroupName' to the Test Application or User with ID '$TestApplicationId'"
+        $ownerAssignment = New-AzRoleAssignment `
+                            -RoleDefinitionName "Owner" `
+                            -ObjectId "$TestApplicationOId" `
+                            -ResourceGroupName "$ResourceGroupName" `
+                            -ErrorAction SilentlyContinue
+
+        if ($ownerAssignment.RoleDefinitionName -eq 'Owner') {
```
Comment (jsquire) [inline eng/common/TestResources/New-TestResources.ps1:712]: Should we write the name if this is an interactive user?   That may make it easier to debug/troubleshoot locally.

File: eng/common/TestResources/New-TestResources.ps1 [hunk 2]
```diff
@@ -686,11 +686,11 @@ try {
...
     $PSBoundParameters['TestApplicationSecret'] = $TestApplicationSecret
 
-    # If the role hasn't been explicitly assigned to the resource group and a cached service principal is in use,
+    # If the role hasn't been explicitly assigned to the resource group and a cached service principal or user authentication is in use,
     # query to see if the grant is needed.
-    if (!$resourceGroupRoleAssigned -and $AzureTestPrincipal) {
+    if (!$resourceGroupRoleAssigned -and $TestApplicationOid) {
         $roleAssignment = Get-AzRoleAssignment `
-                            -ObjectId $AzureTestPrincipal.Id `
+                            -ObjectId TestApplicationOid `
```
Comment (jsquire) [inline eng/common/TestResources/New-TestResources.ps1:693]: I'm assuming this works with either the interactive user or SP.

**Review by jsquire — APPROVED** [review]

File: eng/common/TestResources/New-TestResources.ps1 (hunk 2 above)
Comment (JoshLove-msft) [reply eng/common/TestResources/New-TestResources.ps1]: Yes

File: eng/common/TestResources/New-TestResources.ps1 [hunk 3]
```diff
@@ -619,9 +619,11 @@ try {
             Write-Warning "The specified TestApplicationId '$TestApplicationId' will be ignored when UserAuth is set."
         }
 
-        $TestApplicationOid = (Get-AzADUser -UserPrincipalName (Get-AzContext).Account).Id
+        $userAccount = (Get-AzADUser -UserPrincipalName (Get-AzContext).Account)
+        $TestApplicationOid = $userAccount.Id
         $TestApplicationId = $testApplicationOid
-        Log "User-based app id '$TestApplicationId' will be used."
+        $userAccountName = $userAccount.UserPrincipalName
+        Log "User authentication with user '$userAccountName'('$TestApplicationId') will be used."
```
Comment (heaths) [inline eng/common/TestResources/New-TestResources.ps1:626]: Nit:
```suggestion
        Log "User authentication with user '$userAccountName' ('$TestApplicationId') will be used."
```

File: eng/common/TestResources/New-TestResources.ps1 [hunk 4]
```diff
@@ -702,19 +704,20 @@ try {
...
-        Log "Attempting to assigning the 'Owner' role for '$ResourceGroupName' to the Test Application '$TestApplicationId'"
-        $principalOwnerAssignment = New-AzRoleAssignment `
-                                        -RoleDefinitionName "Owner" `
-                                        -ApplicationId "$TestApplicationId" `
-                                        -ResourceGroupName "$ResourceGroupName" `
-                                        -ErrorAction SilentlyContinue
-
-        if ($principalOwnerAssignment.RoleDefinitionName -eq 'Owner') {
-            Write-Verbose "Successfully assigned ownership of '$ResourceGroupName' to the Test Application '$TestApplicationId'"
+        $idSlug = if ($userAuth) { "user '$userAccountName'('$TestApplicationId')"} else { "Test Application '$TestApplicationId'"};
```
Comment (heaths) [inline eng/common/TestResources/New-TestResources.ps1:707]:
```suggestion
        $idSlug = if ($userAuth) { "User '$userAccountName' ('$TestApplicationId')"} else { "Test Application '$TestApplicationId'"};
```

**Review by heaths — APPROVED** [review]
A few nits to look more appealing, but otherwise LGTM.

**Review by benbp — APPROVED** [review]

**Pull request merged**